src/SWP/Bundle/CoreBundle/AnalyticsExport/CsvFileWriter.php

```diff
@@ -0,0 +1,17 @@
+<?php
+
+namespace SWP\Bundle\CoreBundle\AnalyticsExport;
+
+class CsvFileWriter
+{
+    public function write(string $fileSourcePath, array $data): void
+    {
+        $fp = fopen($fileSourcePath, 'w');
```

Comment:
cant we use here flysystem?

Reply:
I think it's an extra dependency which is not needed here because I can do the same using only three PHP built-in functions instead of injecting the whole service.

Reply:
But we shouldn't write directly to filesystem. In multi server env or one like heroku it will fail. This file should be stored in some external storage (like s3) if it's configured.

Reply:
it won't fail. It will save the file in the cache dir in this case (or any temp dir) and read it because it's the same process - at least on Heroku. It's just the temp file that is then uploaded to the S3 if configured. 

In this case, we have to change it in other places too. Because for example when downloading the author's avatar, generating webp format of images, article images, it's also writing to local temp storage and then uploads to s3 if configured or am I missing something?

Reply:
ok, i was thinking that it's finall generated report destination - where link from email points.